# U.S. Medical Insurance Costs

In [3]:
import csv

csv import is done which will be needed to open the file 

In [5]:
ages = []
sexes = []
bmis = []
num_children = []
smoker_statuses = []
regions = []
insurance_charges = []

empy lists has been created as per the csv

In [21]:
with open("insurance.csv") as file:
    csv_reader = csv.DictReader(file)
    for row in csv_reader:
        ages.append(int(row['age']))
        sexes.append(row['sex'])
        bmis.append(float(row['bmi']))
        num_children.append(int(row['children']))
        smoker_statuses.append(row['smoker'])
        regions.append(row['region'])
        insurance_charges.append(float(row['charges']))

above code appends data to each variabe

In [160]:
class Patients:
    def __init__(self, ages, sexes, bmis, num_children, smoker_statuses, regions, charges):
        self.ages = ages
        self.sexes = sexes
        self.bmis = bmis
        self.num_children = num_children
        self.smoker_statuses = smoker_statuses
        self.regions = regions
        self.charges = charges
        
    def count_sex(self):
        male_count=0
        female_count=0
        for sex in self.sexes:
            if sex=='male':
                male_count+=1
            else:
                female_count+=1
        return [male_count,female_count]
    
    def male_female_ratio(self):
        male_count,female_count=self.count_sex()
       
        ratio=male_count/female_count
        print(f"male to female ratio is {ratio:.2f}.")
    
    def avarage_age(self,sex="Both"):
        Usex=sex.upper()
        total=0
        count=0
        if Usex=="BOTH" :
            for age in self.ages:
                total+=age
                count+=1
        elif Usex=="FEMALE":
            for i in range(len(self.ages)):
                if self.sexes[i]=="female":
                    total+=self.ages[i]
                    count+=1
                else:
                    pass
        elif Usex=="MALE":
            for i in range(len(self.ages)):
                if self.sexes[i]=="male":
                    total+=self.ages[i]
                    count+=1
                else:
                    pass
        else:
            pass
        
        avarage=total/count
        return {f"Avarage age is {avarage:.2f} for {Usex} gender"}
    
    def avarage_charge(self,sex="Both"):
        Usex=sex.upper()
        total=0
        count=0
        if Usex=="BOTH" :
            for charge in self.charges:
                total+=charge
                count+=1
        elif Usex=="FEMALE":
            for i in range(len(self.charges)):
                if self.sexes[i]=="female":
                    total+=self.charges[i]
                    count+=1
                else:
                    pass
        elif Usex=="MALE":
            for i in range(len(self.ages)):
                if self.sexes[i]=="male":
                    total+=self.charges[i]
                    count+=1
                else:
                    pass
        else:
            pass
        
        avarage=total/count
        return {f"Avarage charge is {avarage:.2f} for {Usex} gender"}
    
    
    def bmi_category(self):
        bmi_categories = {
            "Underweight": {"total_charge": 0, "count": 0},
            "Normal weight": {"total_charge": 0, "count": 0},
            "Overweight": {"total_charge": 0, "count": 0},
            "Obesity Class I (Moderate)": {"total_charge": 0, "count": 0},
            "Obesity Class II (Severe)": {"total_charge": 0, "count": 0},
            "Obesity Class III (Very severe or morbid obesity)": {"total_charge": 0, "count": 0}
        }

        for bmi, charge in zip(self.bmis, self.charges):
            if bmi < 18.5:
                category = "Underweight"
            elif 18.5 <= bmi <= 24.9:
                category = "Normal weight"
            elif 25.0 <= bmi <= 29.9:
                category = "Overweight"
            elif 30.0 <= bmi <= 34.9:
                category = "Obesity Class I (Moderate)"
            elif 35.0 <= bmi <= 39.9:
                category = "Obesity Class II (Severe)"
            else:  # bmi >= 40.0
                category = "Obesity Class III (Very severe or morbid obesity)"

            bmi_categories[category]["total_charge"] += charge
            bmi_categories[category]["count"] += 1
        return bmi_categories
    


    def average_charge_by_bmi_category(self):
        bmi_categories=self.bmi_category()
        average_charges = {}
        for category, data in bmi_categories.items():
            if data["count"] > 0:
                average_charges[category] = round((data["total_charge"] / data["count"]),2)
            else:
                average_charges[category] = 0

        return average_charges
    
    def bmi_category_charge_percentage(self):
        bmi_categories=self.bmi_category()
        total_patients=len(self.charges)
        total_charge=0
        for charge in self.charges:
            total_charge+=charge
        percentage_ratio={}
        for category, data in bmi_categories.items():
            if category not in percentage_ratio:
                percentage_ratio[category] = {}
            patient_percentage=(data["count"]/total_patients)*100
            charge_percentage=(data["total_charge"]/total_charge)*100
            percentage_ratio[category][f" {patient_percentage:.2f}% of total patient"]= f"Give {charge_percentage:.2f}% of charge"
        return percentage_ratio 
    
    def smoke_relatiton_charge(self):
        smoke={
            "smoker":{"total_charge":0,"count":0},
            "nonsmoker":{"total_charge":0,"count":0}
        
        }
        avarage_smoker={}
        
        for smoker_status,charge in zip(self.smoker_statuses,self.charges):
            if smoker_status=="yes":
                smoke["smoker"]["total_charge"]+=charge
                smoke["smoker"]["count"]+=1
            else:
                smoke["nonsmoker"]["total_charge"]+=charge
                smoke["nonsmoker"]["count"]+=1
        for category,data in smoke.items():
            avarage_smoker[category]=data["total_charge"] / data["count"]
            
        return avarage_smoker
        
        
    def region_status(self):
        regions={}
        total_patients=len(self.charges)
        region_status={}
        total_charge=0
        for charge in self.charges:
            total_charge+=charge
            
        for region,charge in zip(self.regions,self.charges):
            if region not in regions:
                regions[region]={}
                regions[region]["total_charge"]=0
                regions[region]["count"]=0
            else:
                regions[region]["total_charge"]+=charge
                regions[region]["count"]+=1
        for category,data in regions.items():
            if category not in region_status:
                region_status[category]={}
            patient_percentage=(data["count"]/total_patients)*100
            charge_percentage=(data["total_charge"]/total_charge)*100
            region_status[category][f"Population: {patient_percentage:.2f}% "]= f"Charge {charge_percentage:.2f}% "
            
        return region_status
    
    def atleast_one_child(self):
        total_patients=len(self.num_children)
        count=0
        for child in self.num_children:
            if child>=1:
                count+=1
        percentage=(count/total_patients)*100
        
        return f"{percentage:.2f}% of patients has at least one child"
    
    def analyze_age(self):
        Male=patient.avarage_age('male')
        Female=patient.avarage_age('female')
        both=patient.avarage_age()
        
        print(Male)
        print(Female)
        print(both)
    
    def analyze_charge(self):
        Male=patient.avarage_charge('male')
        Female=patient.avarage_charge('female')
        both=patient.avarage_charge()
        
        print(Male)
        print(Female)
        print(both)
        
                    
                
        
        
        
            
                    

In [164]:
patient = Patients(ages, sexes, bmis, num_children, smoker_statuses, regions, insurance_charges)


In [159]:
patient.male_female_ratio()
patient.atleast_one_child()

male to female ratio is 1.02.


'57.10% of patients has at least one child'

Male to Female Ratio
The analysis of the patient demographic data reveals that the male to female ratio is approximately 1.02. This indicates a near-equal distribution of male and female patients within the dataset.

Parental Status
Further examination of the patient data shows that 57.10% of patients have at least one child. This significant proportion suggests that a majority of the patients included in the study are parents.

In [148]:
patient.analyze_age()

{'Avarage age is 38.917160 for MALE gender'}
{'Avarage age is 39.503021 for FEMALE gender'}
{'Avarage age is 39.207025 for BOTH gender'}


The analysis highlights slight variations in average age between male and female genders. Female patients have a slightly higher average age (39.50 years) compared to male patients (38.92 years). When considering both genders collectively, the average age is 39.21 years.

In [162]:
patient.analyze_charge()

{'Avarage charge is 13956.75 for MALE gender'}
{'Avarage charge is 12569.58 for FEMALE gender'}
{'Avarage charge is 13270.42 for BOTH gender'}


The analysis of the average charges reveals notable differences based on gender. Male patients incur higher average charges ($13,956.75) compared to female patients ($12,569.58). When considering the entire patient population, the overall average charge is $13,270.42. These findings suggest a gender disparity in the healthcare charges, which could be attributed to a variety of factors such as differences in treatment types, healthcare needs, or utilization patterns between male and female patients. Understanding these differences is crucial for healthcare providers and policymakers to ensure equitable and efficient allocation of healthcare resources.

In [163]:
patient.average_charge_by_bmi_category()


{'Underweight': 8852.2,
 'Normal weight': 10379.5,
 'Overweight': 10993.99,
 'Obesity Class I (Moderate)': 14298.84,
 'Obesity Class II (Severe)': 17073.19,
 'Obesity Class III (Very severe or morbid obesity)': 16430.78}

The analysis of the average charges based on weight categories reveals a clear trend where healthcare charges increase with higher weight categories. Patients classified as underweight incur the lowest average charges ($8,852.20), whereas those in Obesity Class II (Severe) incur the highest average charges ($17,073.19). This trend suggests that as the severity of obesity increases, so do the healthcare costs, likely due to the increased medical needs and complications associated with higher weight categories. 

In [132]:
patient.bmi_category_charge_percentage()

{'Underweight': {' 1.49% of total patient': 'Give 1.00% of charge'},
 'Normal weight': {' 16.59% of total patient': 'Give 12.98% of charge'},
 'Overweight': {' 28.18% of total patient': 'Give 23.34% of charge'},
 'Obesity Class I (Moderate)': {' 29.00% of total patient': 'Give 31.25% of charge'},
 'Obesity Class II (Severe)': {' 16.67% of total patient': 'Give 21.44% of charge'},
 'Obesity Class III (Very severe or morbid obesity)': {' 8.07% of total patient': 'Give 9.99% of charge'}}

The analysis of insurance charges based on weight categories highlights significant disparities in the distribution of healthcare costs. While only 1.49% of patients are classified as underweight, they contribute just 1.00% of the total insurance charges. In contrast, patients in the "Obesity Class I (Moderate)" category, who constitute 29.00% of the total patient population, account for the highest proportion of total insurance charges at 31.25%.

Patients in the "Obesity Class II (Severe)" category, although comprising 16.67% of the patient population, contribute a substantial 21.44% of the total insurance charges. Similarly, the "Obesity Class III (Very severe or morbid obesity)" group, representing 8.07% of patients, accounts for 9.99% of total insurance charges.

These findings indicate that higher weight categories are associated with disproportionately higher insurance charges, reflecting the increased healthcare needs and associated costs for patients with obesity. 

In [134]:
patient.smoke_relatiton_charge()

{'smoker': 32050.23183153284, 'nonsmoker': 8434.268297856212}

The analysis demonstrates a substantial disparity in average insurance charges between smokers and non-smokers. Smokers have significantly higher average insurance charges, with an average of $32,050.23, compared to non-smokers, whose average insurance charge is $8,434.27.

This stark contrast underscores the considerable impact of smoking on healthcare costs, indicating that smokers tend to incur substantially higher healthcare expenses covered by insurance. 

In [135]:
patient.region_status()

{'southwest': {'Population: 24.25% ': 'Charge 22.55% '},
 'southeast': {'Population: 27.17% ': 'Charge 30.20% '},
 'northwest': {'Population: 24.25% ': 'Charge 22.67% '},
 'northeast': {'Population: 24.18% ': 'Charge 24.45% '}}

The analysis indicates variations in insurance charges across different regions. The Southeast region stands out with both the highest population percentage (27.17%) and the highest insurance charges (30.20%). In contrast, the Southwest and Northwest regions have similar population percentages (24.25%) but slightly lower insurance charges (22.55% and 22.67% respectively).

The Northeast region has a population percentage close to that of the Southwest and Northwest (24.18%), but it exhibits higher insurance charges (24.45%).